### PR TITLE
Add documentation on running full system and simulator

### LINKED ISSUE #1737: Document Visualizing SimulatedTests, running AI, running simulator
### Description of the task
as titled + any other useful bazel commands

### Acceptance criteria

- [ ] as titled

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
Just added a small section on running full_system/simulator/robot diagnostics.

### Testing Done
N/A

### Resolved Issues
resolves #1737 

### Length Justification
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -139,6 +139,17 @@ Now that you're setup, if you can run it on the command line, you can run it in
     4. For `Bazel Command` you can put any bazel command, like `build`, `run`, `test`, etc.
     5. Click `Ok`, then there should be a green arrow in the top right corner by the drop-down menu. Click it and the test will run!
 
+### Running our AI, Simulator or Robot Diagnostics
+
+1. Open your terminal and run `ifconfig`.
+2. Pick the network interface you would like to use:
+    1. If you are running things locally, you can pick any interface that is not `lo`
```
Comment (kelvinkoon) [inline docs/getting-started.md:146]: Just to confirm, on Wednesday's meeting, you mentioned you could run loopback. I assume that was for something else and not this, right?

**Review by kelvinkoon — DISMISSED** [review]
Small question, otherwise LGTM :memo:

File: docs/getting-started.md [hunk 2]
```diff
@@ -139,6 +139,17 @@ Now that you're setup, if you can run it on the command line, you can run it in
...
+    2. If you would like to communicate with robots on the network, make sure to select the interface that is connected to the same network as the robots.
+3. Run AI: `bazel run //software:full_system -- --interface=interface_here --backend=WifiBackend`
+4. Run Simulator: `bazel run //software/simulation:standalone_simulator_main -- --interface=interface_here`
+5. Run Robot Diagnostics: `bazel run //software/gui/robot_diagnostics:robot_diagnostics_main -- --interface=interface_here --backend=WifiBackend`
```
Comment (itsarune) [inline docs/getting-started.md]: Maybe consider including the expected outcomes for each command

File: docs/getting-started.md [hunk 3]
```diff
@@ -139,6 +139,17 @@ Now that you're setup, if you can run it on the command line, you can run it in
     4. For `Bazel Command` you can put any bazel command, like `build`, `run`, `test`, etc.
     5. Click `Ok`, then there should be a green arrow in the top right corner by the drop-down menu. Click it and the test will run!
 
+### Running our AI, Simulator or Robot Diagnostics
+
+1. Open your terminal and run `ifconfig`.
+2. Pick the network interface you would like to use:
+    1. If you are running things locally, you can pick any interface that is not `lo`
+    2. If you would like to communicate with robots on the network, make sure to select the interface that is connected to the same network as the robots.
+3. Run AI: `bazel run //software:full_system -- --interface=interface_here --backend=WifiBackend`
```
Comment (jonathanlew) [inline docs/getting-started.md]: nit:
```suggestion
3. Run AI: `bazel run //software:full_system -- --interface=[interface_here] --backend=WifiBackend`
```

and same for below, but it's a personal preference thing, so your call

File: docs/getting-started.md [hunk 4]
```diff
@@ -139,6 +139,17 @@ Now that you're setup, if you can run it on the command line, you can run it in
     4. For `Bazel Command` you can put any bazel command, like `build`, `run`, `test`, etc.
     5. Click `Ok`, then there should be a green arrow in the top right corner by the drop-down menu. Click it and the test will run!
 
+### Running our AI, Simulator or Robot Diagnostics
```
Comment (jonathanlew) [inline docs/getting-started.md:142]: Can we add a description of what running AI/Simulator/Robot Diagnostics does? Like why would someone want to do this?

Edit: this is similar to @itsarune's comment

**Review by jonathanlew — CHANGES_REQUESTED** [review]

File: docs/getting-started.md (hunk 3 above)
Comment (akhilveeraghanta) [reply docs/getting-started.md]: fixed

File: docs/getting-started.md (hunk 4 above)
Comment (akhilveeraghanta) [reply docs/getting-started.md:142]: added

File: docs/getting-started.md (hunk 2 above)
Comment (akhilveeraghanta) [reply docs/getting-started.md]: good point, added!

File: docs/getting-started.md (hunk 1 above)
Comment (akhilveeraghanta) [reply docs/getting-started.md:146]: yeah loop back doesn't work :'( i tried it, it works for a bit and then crashes

**Comment by akhilveeraghanta** [comment]
I honestly thought that this PR got merged, completely forgot about it my bad, fixed everyone's comments ready for another round of review!

**Review by itsarune — DISMISSED** [review]

File: docs/getting-started.md [hunk 5]
```diff
@@ -139,6 +139,24 @@ Now that you're setup, if you can run it on the command line, you can run it in
...
+    1. If you are running things locally, you can pick any interface that is not `lo`
+    2. If you would like to communicate with robots on the network, make sure to select the interface that is connected to the same network as the robots.
+3. Run our AI: `bazel run //software:full_system -- --interface=[interface_here] --backend=WifiBackend`
+    - This will launch the Visualizer, which displays what the AI is currently "seeing" and allows us to interact with the AI through the dynamic parameters.
+    - The field should be empty, as we are currently not receiving SSL Vision packets.
+4. Run our Simulator: `bazel run //software/simulation:standalone_simulator_main -- --interface=[interface_here]`
+    - The Simulator runs our firmware and Box2D (a physics engine) to simulate how our robots would behave on the field.
+    - The Simulator outputs SSL Vision packets, which contain position information of all robots and the ball.
+    - Our AI can now "see" the robots, and they should be displayed on the Visualizer.
+    - You can use cntrl-click to move the ball around in the Simulator, and try changing the Play Override on the Visualizer to see the robots move!
```
Comment (kelvinkoon) [inline docs/getting-started.md]: Is "cntrl-click" the intended phrase?

**Review by kelvinkoon — CHANGES_REQUESTED** [review]
Last question, otherwise LGTM

**Review by jonathanlew — COMMENTED** [review]
LGTM pending other reviewers' comments

**Review by jonathanlew — DISMISSED** [review]

File: docs/getting-started.md [hunk 6]
```diff
@@ -139,6 +139,24 @@ Now that you're setup, if you can run it on the command line, you can run it in
     4. For `Bazel Command` you can put any bazel command, like `build`, `run`, `test`, etc.
     5. Click `Ok`, then there should be a green arrow in the top right corner by the drop-down menu. Click it and the test will run!
 
+### Running our AI, Simulator or Robot Diagnostics
+
+1. Open your terminal and run `ifconfig`.
+2. Pick the network interface you would like to use:
+    1. If you are running things locally, you can pick any interface that is not `lo`
+    2. If you would like to communicate with robots on the network, make sure to select the interface that is connected to the same network as the robots.
```
Comment (tarakong01) [inline docs/getting-started.md:147]: How do we know which interface is connected to the robots? This is more of a general question for me as I'm not too sure how the different interfaces are used

File: docs/getting-started.md (hunk 5 above)
Comment (akhilveeraghanta) [reply docs/getting-started.md]: fixed!

File: docs/getting-started.md (hunk 6 above)
Comment (akhilveeraghanta) [reply docs/getting-started.md:147]: Usually most computers only have 1 or 2 interfaces. You can run `ifconfig` and you will see all of the available interfaces on the left column. The one with `lo` is the loopback interface which is internal to your computer so you never want to use that one. 

The other two (you may only have 1 for WiFi if your computer doesn't have ethernet) you can infer from the interface name. 

Usually wlp3s0, or wlan0 are used for wireless interfaces and enp6s0, eth0 are used for wired interfaces (the numbers can be different). If you have more than 1, and they are not clear from the name, you can just use trial and error at that point, you won't see the robots in the visualizer if the interface doesn't work

**Review by jonathanlew — APPROVED** [review]

**Review by kelvinkoon — APPROVED** [review]

**Review by tarakong01 — APPROVED** [review]

**Review by itsarune — APPROVED** [review]

**Pull request merged**